### PR TITLE
feat(wasm): support for NPM and Browser

### LINKED ISSUE #1038: feat: publish tailcall WASM build on NPM
Publish tailcall on npm under 

- `@tailcallhq/wasm-browser`
- `@tailcallhq/wasm-node`

**Public API**

```js
import {GraphQLExecutor} from '@tailcallhq/tailcall'


// YML, JSON or GraphQL
const jsonConfig = " ... <tailcall config> ... "; 


// Use the builder to create the configuration
const builder = new TailcallBuilder().withJSONConfig(jsonConfig).withYMLConfig(ymlConfig);

// Build an instance of Tailcall, resolving configurations
const tailcall = await builder.build();


// Execute queries using the builder.
const query = "query {users {name}}"
await executor.execute({query})
```

**Technical Requirements**
- I should be able to install the package via npm
- I should be able to use the package in the browser directly
- I should be able to use the package via node.js

### PR DESCRIPTION
**Pull request by ssddOnTop**
**Issue Reference(s):**  
Fixes #1038
/claim 1038

**TODO**
- [x] add tests
- [x] fix logging
- [x] fix cache

### CODE REVIEW

File: tailcall-wasm/.cargo/config.toml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[target.wasm32-unknown-unknown]
+runner = 'wasm-bindgen-test-runner'
+
+[build]
+target = "wasm32-unknown-unknown"
```
Comment (tusharmath) [inline tailcall-wasm/.cargo/config.toml:5]: Can you check if we can move this logic into `tailcall-wasm/cargo.toml`

File: tailcall-wasm/README.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+### NodeJS
+
+```javascript
+const tc = require("@tailcallhq/tailcall-node")
+
+async function run() {
+  try {
+    let schema = "https://raw.githubusercontent.com/tailcallhq/tailcall/main/examples/jsonplaceholder.graphql"
+    let builder = new tc.TailcallBuilder()
```
Comment (tusharmath) [inline tailcall-wasm/README.md:80]:
```js
let b = new tc.TailcallBuilder();

let exe = await b
  .withJsonConfig(" some json config>  ")
  .withYAMLConfig(" <some yml config>  ")
  .withConfigPath("https://raw.githubusercontent.com/tailcallhq/tailcall/main/examples/jsonplaceholder.graphql")
  .build();
```

Something like the above API would be better. 

We should not pass the content of the file and also the file path ever.

Comment (ssddOnTop) [reply tailcall-wasm/README.md:80]: The problem with that is, that we won't be able to access the code from ConfigReader, we can serialize Config but resolving that might be a problem.

To save memory, we can drop all the detected config files after building the Executor.

**Comment by tusharmath** [comment]
Unable to run Tailcall via the browser. 

Moving to draft to reduce noise and improve CI efficiency. Once you are ready just mark it as "ready to review". Feel free to give a shoutout on the #contributors channel on Discord if you want immediate attention.

**Comment by ssddOnTop** [comment]
> Unable to run Tailcall via the browser.

Updated docs..

after running python server, you need to pass a query `?config=<>` 
example: 

```url
http://0.0.0.0:8000/example/browser/?config=https://raw.githubusercontent.com/tailcallhq/tailcall/main/examples/jsonplaceholder.graphql
```

File: tailcall-wasm/package.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+  "scripts": {
+    "compile-node": "wasm-pack build --release --target nodejs --scope tailcallhq --out-dir ./node/pkg",
+    "compile-browser": "wasm-pack build --release --target web --scope tailcallhq --out-dir ./browser/pkg",
+    "dev-node": "wasm-pack build --dev --target nodejs --scope tailcallhq --out-dir ./node/pkg",
+    "dev-browser": "wasm-pack build --dev --target web --scope tailcallhq --out-dir ./browser/pkg",
+    "publish": "npm run compile-node && wasm-pack publish --target nodejs"
```
Comment (tusharmath) [inline tailcall-wasm/package.json:7]:
```suggestion
    "build-release-node": "wasm-pack build --release --target nodejs --scope tailcallhq --out-dir ./node/pkg",
    "build-release-browser": "wasm-pack build --release --target web --scope tailcallhq --out-dir ./browser/pkg",
    "build-dev-node": "wasm-pack build --dev --target nodejs --scope tailcallhq --out-dir ./node/pkg",
    "build-dev-browser": "wasm-pack build --dev --target web --scope tailcallhq --out-dir ./browser/pkg",
    "publish": "npm run compile-node && wasm-pack publish --target nodejs"
```

**Review by tusharmath — APPROVED** [review]

**Pull request merged**